scylla/src/transport/session.rs

```diff
@@ -53,6 +53,10 @@ pub use crate::transport::connection_pool::PoolSize;
 #[cfg(feature = "ssl")]
 use openssl::ssl::SslContext;
 
+pub trait AddressTranslator: Send + Sync {
+    fn translate_address(&self, addr: &SocketAddr) -> SocketAddr;
```

Comment:
What should happen when we come across an address that doesn't have a translation?
Should the driver panic?
Maybe it would be better to return a `Result` here.
This way we could detect that the provided translation information is not sufficient and e.g. log it so that the user can fix it.

Reply:
I get your point, the described use case is convincing. Will change that.

Reply:
> How do other drivers handle this issue (e.g. java-driver)?

java-driver returns bare address, and does not even bother with checking it for nullity.
```
public class MyAddressTranslator implements AddressTranslator {
    public InetSocketAddress translate(InetSocketAddress address) {
        ... // your custom translation logic
    }
}
```